application/src/main/java/org/thingsboard/server/service/install/SqlDatabaseUpgradeService.java

```diff
@@ -480,7 +488,41 @@ public void upgradeDatabase(String fromVersion) throws Exception {
                     } catch (Exception e) {
                         log.error("Failed updating schema!!!", e);
                     }
+                  
+                    deviceProfileRepository.findAll().forEach(deviceProfile -> {
```

Comment:
findAll is very aggressive. We have environments with more than 50K device profiles. Please use pagination. 
Move this code to DefaultDataUpdateService (see tenantsDefaultRuleChainUpdater or similar method there)